### ISSUE TITLE
"onBackdropPress" runs when keyboard is open

### ISSUE THREAD

**Issue by yasir-netlinks** [react-native-modal/react-native-modal#244]
### Info
react-native => "^0.57.8"
react-native-modal => "^7.0.2"
testing on "Samsung S9+"

### Problem
I am testing on android, while everything functions properly when there is no keyboard, but when keyboard is open, the `onBackdropPress` executes when clicked inside the modal as shown below in the image:
![imageedit_2_8435245230](https://user-images.githubusercontent.com/26054956/51891705-8180be00-23bd-11e9-94c2-7ae4daf0f0b8.jpg)

This is the style of the modal container:  
```
container: {
        position: 'relative',
        padding: moderateScale(20),
        alignItems: 'flex-start',
        backgroundColor: '#fff',
        borderWidth: 1,
        borderColor: COLOR_BORDER_LIGHT,
        borderRadius: 30,
}
```

**Comment by mmazzarolo** [comment]
🤔 you mean by pressing on the white padding or even on the content?

**Comment by yasir-netlinks** [comment]
@mmazzarolo , sorry for the late reply, yes the white padding, if I tap on the lower part of the submit button, the modal gets dismissed, not sure if anyone else had this problem!

**Comment by mmazzarolo** [comment]
Seems to be more of an android layout issue. Mind creating a simple repo for reproducing it? Thank you!

**Comment by danielnmai** [comment]
Can confirm this behavior. When `onBackdropPress` is set to closing the modal, when the keyboard opens up, you can touch anywhere outside of the keyboard to close the modal. It's not a good behavior.

**Comment by mmazzarolo** [comment]
@danielnmai thanks for reporting. 
I guess it could be solved by disabling `onBackdropPress` when the keyboard is open. 
It can be done directly in your app (just put the `onBackdropPress` in the state and change it based on the keyboard state), but the most elegant solution is probably putting it directly in `react-native-modal`. Anyone willing to tackle it with a PR?

**Comment by arielrodrigues** [comment]
@mmazzarolo @danielnmai This issue still happening? I would like to help with this but I have no success reproducing the problem.

**Comment by danielnmai** [comment]
yes, it's still happening. Let's me pull down an example code ASAP.

**Comment by danielnmai** [comment]
Finally have to some time to set up a repo to demonstrate this issue.

First, clone and install this example app here: https://github.com/danielnmai/RNModalDemo
Steps to reproduce the issue:

1. Press the "Show Modal" link on top left corner
2. Press the text input field to bring up the keyboard
3. While the keyboard is open, press anywhere in the screen to close the modal, INCLUDING pressing on the modal itself

Observed behavior: pressing anywhere on the screen, including on the modal, will close the modal
Expected behavior: pressing anywhere outside the screen, NOT including the modal, would close it.
Video: https://d.pr/free/v/WTbtC2

**Comment by souzaluiz** [comment]
is anyone still having this problem? here's a snack based on @danielnmai code [example](https://snack.expo.dev/@souzaluiz/on-backdrop-press-close), I tested it here and couldn't replicate the problem.
